storage/grpc_client.go

```diff
@@ -1688,6 +1691,9 @@ func (w *gRPCWriter) uploadBuffer(recvd int, start int64, doneReading bool) (*st
 				}
 				sent = int(writeOffset) - int(start)
 
+				// Drop the stream reference as a new one will need to be created.
+				w.stream = nil
```

Comment:
Shouldn't this happen outside of the `if ShouldRetry()` conditional?

Reply:
I moved this line